### PR TITLE
Added emoji page

### PR DESCRIPTION
**Pull request by Henonicks**
[ Please outline your change here. Be sure to check the checkboxes below. ]

## Documentation change checklist

- [X] My documentation changes follow the [docs style guide](https://dpp.dev/docs_standards.html) and any code examples follow the [coding style guide](https://dpp.dev/coding-standards.html).
- [X] I tested that my change works before raising the PR (via running `doxygen`, and testing examples).
- [X] I have not moved any existing pages or changed any existing URLs without strong justification as to why.
- [X] I have not generated content using AI or a desktop utility such as grammarly.

### CODE REVIEW

File: docpages/example_programs/misc/using_emoji.md
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using_emoji.md:1]: `kebab-case` for file names (don't change `using_emoji.cpp`, that's okay) and page names please!

*I do know I told you to do `snake_case` before, sorry about that!*

File: docpages/example_code/using_emoji.cpp
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_code/using_emoji.cpp:1]: wahhh they are already tabs what the heck let me see

File: docpages/example_programs/misc/using_emoji.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+\page using_emoji Using emoji
+
+Need your bot to use emoji? There is `dpp::emoji` class for you. It can store either a custom or a unicode emoji's name and ID. More later
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_programs/misc/using_emoji.md:3]: What about "Then the `dpp::emoji` class is just for you."?

File: docpages/example_programs/misc/using_emoji.md
Comment (Henonicks) [reply docpages/example_programs/misc/using_emoji.md:1]: kebab for pages and snake for cpp files?

Comment (Jaskowicz1) [reply docpages/example_programs/misc/using_emoji.md:1]: Yep!

File: docpages/example_programs/misc/using_emoji.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_programs/misc/using_emoji.md:3]: Yeah that could work :)

**Comment by Jaskowicz1** [comment]
If you could add an image at the end of the page, showing the results, that would be great!

**Comment by Henonicks** [comment]
> If you could add an image at the end of the page, showing the results, that would be great!

That was in my plans, I just wanted to make this draft

**Comment by Jaskowicz1** [comment]
> > If you could add an image at the end of the page, showing the results, that would be great!
> 
> That was in my plans, I just wanted to make this draft

Ah rightio, no worries just wanted to make sure :)

**Comment by Henonicks** [comment]
Gotta fix the top of the page

File: docpages/example_code/using_emoji.cpp
Comment (Jaskowicz1) [inline docpages/example_code/using_emoji.cpp:1]: This is still space indentation

File: docpages/example_code/using_emoji.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        if (event.msg.content == "WHAT?") {
+            /* If some unknown content shocks the user */
+
+            bot.message_add_reaction(event.msg.id, event.msg.channel_id, shocked.format());
+            /* React to their message with a shocked guy */
+        } else if (event.msg.content == "I'm hungry") {
+            /* But if they're hungry */
+
+            bot.message_add_reaction(event.msg.id, event.msg.channel_id, dpp::unicode_emoji::cut_of_meat);
+            /* Let's send some meat to the message, so they don't starve. They will thank us later. */
```
Comment (Jaskowicz1) [inline docpages/example_code/using_emoji.cpp:23]: I would put this above the `bot.message_add_reaction`.

File: docpages/example_programs/misc/using-emoji.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+\page using-emoji Using Emoji
+
+Need your bot to use an emoji? Then the `dpp::emoji` class is just for you. It can store either a custom or a unicode emoji's name and ID. To use one in a message or when reacting to one, you use its mention. In select menus, it's the emoji object itself. Here's how to use them:
```
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using-emoji.md:3]: Tbh, the more I think of `Then the dpp::emoji class is just for you`, the more I think it doesn't work. This comes across that `dpp::emoji` should be used in all instances.

This text should be reworded to state both `dpp::emoji` and `dpp::unicode_emoji` and what the difference is.

The `In select menus, it's the emoji object itself.` part I'd put in a note, people may not read this text fully so a separate note is handy for this.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_code/using_emoji.cpp (hunk 2 above)
Comment (Henonicks) [reply docpages/example_code/using_emoji.cpp:23]: But there's a comment below the `if` and I can't put it above the statement

File: docpages/example_programs/misc/using-emoji.md (hunk 3 above)
Comment (Henonicks) [reply docpages/example_programs/misc/using-emoji.md:3]: Like I said, this part needs fixing

File: docpages/example_code/using_emoji.cpp (hunk 2 above)
Comment (Jaskowicz1) [reply docpages/example_code/using_emoji.cpp:23]: that's fine, it makes sense to bunch them together. You could even put them in the same comment and that still makes sense!

File: docpages/example_programs/misc/using-emoji.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+\page using-emoji Using Emoji
+
+Need your bot to use an emoji?
+Then the `dpp::emoji` class is just for you. It can store either a custom or a unicode emoji's name and ID. To use one in a message, you use its mention. To react, there's other kind of format: if you want to use an animated one, it's a:<name>:id, if it's static — <name>:id, if that's a unicode one — the corresponding character. Here, <name> represents the name of the emoji. In select menus, it's the emoji object itself. Here's how to use them:
```
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using-emoji.md:4]: This doesn't really make sense to be on the next line. It's part of the same paragraph.

Again, I think you still need to reword this as it's still making it sound like you should use dpp::emoji when using unicode emojis.

I would start it off with `Emojis are formed in two different, dpp::emoji and dpp::unicode_emoji. dpp::emoji is more commonly used for custom emojis, whereas dpp::unicode_emoji is used for the default emojis. Using dpp::unicode_emoji is pretty simple, but dpp::emoji is a bit more than that.`

Then you could have another paragraph where you start talking about how you add them in a message and then cut your examples into two, with an example here showing only that.

Then do a reactions part, with another example here with only reactions.

Then do the select menus part here.

Have a look at something like https://dpp.dev/setting_status.html, I seperate the examples into two, first example shows "hey, this is how you simply set the status", I then talk about doing it every x seconds and then show an example for that.

Also, `there's other kind of format` is incorrect, it should be `Reactions are a bit more complicated. For example, if you want to use an animated one.....`

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/misc/using-emoji.md (hunk 4 above)
Comment (Mishura4) [reply docpages/example_programs/misc/using-emoji.md:4]: I don't like this either @Jaskowicz1 

dpp::unicode_emoji is not a class but a namespace, it's very different from dpp::emoji
The two should not be equated like that

Comment (Jaskowicz1) [reply docpages/example_programs/misc/using-emoji.md:4]: Hm true @Mishura4

The current text definitely needs to be split off, allowing three different examples

Comment (Henonicks) [reply docpages/example_programs/misc/using-emoji.md:4]: What did I miss 💀

Comment (Henonicks) [reply docpages/example_programs/misc/using-emoji.md:4]: > This doesn't really make sense to be on the next line. It's part of the same paragraph...
Allow me to yoink that

**Comment by Henonicks** [comment]
I have to test indentation

**Comment by Henonicks** [comment]
It works now, just have to push it

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]
If your PR is ready for review, please unmark it as draft, that will help us know that this review was wanted.

File: docpages/example_programs/misc/using-emoji.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+\page using-emoji Sending Emoji
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_programs/misc/using-emoji.md:1]: And also "Using", not "Sending". Yes, got it

File: docpages/example_programs/misc/using-emoji.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+Yay, our bot has emotions now!
+
+\image html using_emoji2.png
+
+Finally, select menus. They're lists that let you select an option with an emoji each and are an optional part of a message. These guys are covered \ref components3 "here". They require emoji components (name, ID, animated state) to come separately. Also, if the emoji you're using isn't animated, you don't have to specify that. If your emoji is unicode, the ID is optional, too, since both are defaulted to none (0/false).
+
+\include{cpp} using_emoji3.cpp
+
+The Cook appears.
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_programs/misc/using-emoji.md:25]: But yeah I should change that, it really doesn't make sense

[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_programs/misc/using-emoji.md:25]: Ah, yeah I think people would be a bit confused by it lol

File: docpages/example_code/using_emoji1.cpp [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+	bot.on_log(dpp::utility::cout_logger());
+
+	/* We'll be using two emojis: shocked guy and animated mad face. */
+	dpp::emoji shocked("vahuyi", 1179366531856093214);
+	dpp::emoji mad("mad", 1117795317052616704, dpp::e_animated); /* We need this third argument, which is an emoji flag. */
+
+	bot.on_slashcommand([shocked, mad](const dpp::slashcommand_t& event) {
+		if (event.command.get_command_name() == "send-emojis") {
+
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_code/using_emoji1.cpp:15]: 15?

[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_code/using_emoji1.cpp:15]: Yes

File: docpages/example_programs/misc/using-emoji.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+\page using-emoji Sending Emoji
+
+Need your bot to use emoji? Then you've come to the right place! Here are shown 3 cases of where and how you would use emoji.
+
+First - sending emoji. You have to use its mention, which depends on the type. If it's a default emoji, you use the corresponding character. So, for example, if you wanted to send a nerd emoji, you would use the nerd unicode character. Now, custom emoji. There are two types: static and animated. Their mentions are `<:[name]:[id]>` and `<a:[name]:[id]>` consequently, where `[name]` means the emoji name and `[id]` is for its ID. When you send such mention, it automatically gets converted into your emoji. Here's an example of sending emojis:
```
[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_programs/misc/using-emoji.md]: Last comment got messed up, but yes, this isn't resolved still. Please look to change  `First - sending emoji` -> `First - Sending emojis`.

File: docpages/example_code/using_emojis1.cpp [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+		if (event.command.get_command_name() == "send-emojis") {
+			/* Here we send our very informative message: three epic emojis. */
+			event.reply(dpp::unicode_emoji::nerd + shocked.get_mention() + mad.get_mention());
+		}
+	});
+
+	bot.on_ready([&bot](const dpp::ready_t& event) {
+		if (dpp::run_once<struct register_bot_commands>()) {
+			dpp::slashcommand send("send-emojis", "Send the emojis", bot.me.id);
+			bot.global_command_create(send);
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_code/using_emojis1.cpp:23]: I'll do that in 3.cpp, too then

File: docpages/example_code/using_emojis3.cpp [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+	});
+
+	bot.on_select_click([](const dpp::select_click_t& event) {
+		event.reply(event.values[0]);
+	});
+
+	bot.on_ready([&bot](const dpp::ready_t& event) {
+		if (dpp::run_once<struct register_bot_commands>()) {
+			dpp::slashcommand select("select", "Send the select menu", bot.me.id);
+			bot.global_command_create(select);
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_code/using_emojis3.cpp:39]: I'm confused

**Review by Jaskowicz1 — APPROVED** [review]

File: docpages/example_programs/misc/using-emojis.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+\image html using_emojis2.png
+
+Finally, select menus. These guys are covered \ref components3 "here". They require emoji components (name, ID, animated state) to come separately. If the emoji you're using isn't animated, you don't have to specify that. If your emoji is unicode, it doesn't even have an ID, so you only put the character, since both animated state and ID are defaulted to none (false/0).
+
+\include{cpp} using_emojis3.cpp
+
+Yay, our context menu is now interesting!
+
+\image html using_emojis3.png
```
Comment (Mishura4) [inline docpages/example_programs/misc/using-emojis.md:27]: Need the empty line here pls

File: docpages/example_programs/misc/using-emojis.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+\page using-emojis Using Emojis
+
+Need your bot to use an emoji? Then you've come to the right place! Here are three examples of using emojis.
+
+First - Sending emojis. You have to use its mention, which depends on the type. If it's a default emoji, you use the corresponding character. So, for example, if you wanted to send a nerd emoji, you would use the nerd unicode character. Now, custom emoji. There are two types: static and animated. Their mentions are `<:[name]:[id]>` and `<a:[name]:[id]>`, where `[name]` means the emoji name and `[id]` is for its ID. When you send such mention, it automatically gets converted into your emoji. Here's an example of sending emojis:
```
Comment (Mishura4) [inline docpages/example_programs/misc/using-emojis.md:5]: Should probably mention that the bot needs access to the emoji, which means the bot being on the server the emojis are from, and permissions for the server to use them on

**Review by Mishura4 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/misc/using-emojis.md (hunk 12 above)
Comment (Henonicks) [reply docpages/example_programs/misc/using-emojis.md:5]: \note?

File: docpages/example_programs/misc/using-emojis.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+\page using-emojis Using Emojis
+
+Need your bot to use an emoji? Then you've come to the right place! Here are three examples of using emojis.
+
+\note If your bot isn't in a guild where you want to take the custom emoji from, it won't work. If you're trying to send one, it'll turn into `:name:`, otherwise it won't work at all giving you the Unknown Emoji error.
```
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using-emojis.md:5]: The wording here is a bit odd. Correct to:
```
\note If your bot isn't in the guild where you want to use the custom emoji from, it won't work, giving you dpp::exception_error_code::err_unknown_emoji.
```

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/misc/using-emojis.md (hunk 13 above)
Comment (Henonicks) [reply docpages/example_programs/misc/using-emojis.md:5]: Yeah I'll just yoink that

Comment (Jaskowicz1) [reply docpages/example_programs/misc/using-emojis.md:5]: That was the point lol

File: docpages/example_programs/misc.md [hunk 14]
```diff
@@ -11,3 +11,4 @@ This section lists examples that do not fit neatly into any of the categories ab
 * \subpage cpp-eval-command-discord
 * \subpage checking-member-permissions
 * \subpage setting_status
+* \subpage usinfg-emojis
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_programs/misc.md]: Akcshually tested it this time

File: docpages/example_programs/misc/using-emojis.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
+\page using-emojis Using Emojis
+
+Need your bot to use an emoji? Then you've come to the right place! Here are three examples of using emojis.
+
+\note If your bot isn't in the guild where you want to use the custom emoji from, it won't work, giving you `dpp::exception_error_code::err_unknown_emoji`.
```
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using-emojis.md:5]: Please take `dpp::exception_error_code::err_unknown_emoji` out of the code-block so it links correctly (allowing people to click it to view it)

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/misc/using-emojis.md (hunk 15 above)
Comment (Henonicks) [reply docpages/example_programs/misc/using-emojis.md:5]: Also the code block is not the problem

Comment (Jaskowicz1) [reply docpages/example_programs/misc/using-emojis.md:5]: True, not sure why your branch isn’t showing it linking. Either way, we’ve stated this several time and I’ll update the docs convention for it, you do not put things like dpp::cluster and whatnot in code blocks as we all agreed it looks awful when linked inside a code block.

**Comment by Jaskowicz1** [comment]
Just gonna investigate why `dpp::exception_error_code::err_unknown_emoji` isn't linking on the preview before I approve.

File: docpages/example_programs/misc/using-emojis.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
+\page using-emojis Using Emojis
+
+Need your bot to use an emoji? Then you've come to the right place! Here are three examples of using emojis.
+
+\note If your bot isn't in the guild where you want to use the custom emoji from, it won't work, giving you dpp::exception_error_code::err_unknown_emoji.
```
Comment (Jaskowicz1) [inline docpages/example_programs/misc/using-emojis.md:5]: Figured it out, change `dpp::exception_error_code::err_unknown_emoji` to `dpp::err_unknown_emoji` (without codeblock ofc). Not sure why doxygen isn't accepting it with the current state but oh well.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/misc/using-emojis.md [hunk 17]
```diff
@@ -0,0 +1,29 @@
+\page using-emojis Using Emojis
+
+Need your bot to use an emoji? Then you've come to the right place! Here are three examples of using emojis.
+
+\note If your bot isn't in the guild where you want to use the custom emoji from, it won't work, giving you dpp::err_unknown_emoji.
```
Comment (Mishura4) [inline docpages/example_programs/misc/using-emojis.md:5]: `isn't in the guild you want to use the custom emoji from`**

**Review by Mishura4 — CHANGES_REQUESTED** [review]

**Review by Mishura4 — APPROVED** [review]
